### PR TITLE
docs: Add example for using ALB with GRPC

### PR DESCRIPTION
**Pull request by backjo**
Add example for using ALB Ingress with GRPC, since ALBs support GRPC backends now!

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: docs/operator-manual/ingress.md [hunk 1]
```diff
@@ -351,13 +351,62 @@ spec:
...
+  selector:
+    app.kubernetes.io/name: argocd-server
+  sessionAffinity: None
+  type: ClusterIP
+```
+
+Once we create this service, we can configure the Ingress to conditionally route all `application/grpc` traffic to the new HTTP2 backend, using the `alb.ingress.kubernetes.io/conditions` annotation, as seen below. Note: The value after the . in the condition annotation _must_ be the same name as the service that you want traffic to route to - and will be applied on any path with a matching serviceName. 
+
+```yaml
+  apiVersion: extensions/v1beta1
```
Comment (jannfis) [inline docs/operator-manual/ingress.md:383]: Recent versions of K8s moved `Ingress` to `networking.k8s.io/v1` API. Maybe this should be reflected in a comment, so people C&P this YAML will have no bad surprise.

File: docs/operator-manual/ingress.md [hunk 2]
```diff
@@ -351,13 +351,62 @@ spec:
...
+            serviceName: argogrpc
+            servicePort: 443
+          pathType: ImplementationSpecific
+        - backend:
+            serviceName: argocd-server
+            servicePort: 443
+          pathType: ImplementationSpecific
+    tls:
+    - hosts:
+      -argocd.argoproj.io
```
Comment (jannfis) [inline docs/operator-manual/ingress.md:409]: I think there's a missing whitespace between dash and `argocd.argoproj.io`

**Review by jannfis — CHANGES_REQUESTED** [review]
Thanks a lot for improving documentation, @backjo!

Looks good to me so far, with just a typo and one proposal for improvement up for discussion. Please see inline comments.

File: docs/operator-manual/ingress.md (hunk 1 above)
Comment (backjo) [reply docs/operator-manual/ingress.md:383]: ah yes, missed that!

**Review by jannfis — APPROVED** [review]
Awesome, thank you for updating!

LGTM!

**Comment by sidewinder12s** [comment]
@backjo Hey I finally got around to trying to implement proper gRPC support as well, but ran into a slight difference from the docs I was wondering if you might have run into as well. 

Main one was switching the service backend protocol explicitly to gRPC:

```yaml
alb.ingress.kubernetes.io/backend-protocol-version: GRPC
```
It appears the http2 annotation works, but the health check fails until you switch it to gRPC where it then does some specific logic for health checking grpc services. Otherwise the target group shows unhealthy, but everything appears to work anyways.

**Comment by barektui** [comment]
> Main one was switching the service backend protocol explicitly to gRPC

I wish we could have a better documentation. Thank you for pointing that. What i'd like to say is to be careful with:

```
alb.ingress.kubernetes.io/healthcheck-path: "/healthz"
```
What i expected is to change https backend healthcheck path. What i get is breaking up grpc backend healthcheck. Btw - defaults works.

**Pull request merged**